agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/AgentImplementation.java

```diff
@@ -153,6 +153,8 @@ public static String getAgentJarLocation() throws UnsupportedEncodingException {
         }
 
         String path = AgentImplementation.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+        int index = path.lastIndexOf('/');
+        path = path.substring(0, index + 1);
```

Comment:
check that index is not `-1`. Just in case so code will not throw. Also see above - using Path instead of string will help